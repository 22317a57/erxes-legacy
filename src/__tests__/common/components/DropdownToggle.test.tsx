import { mount, shallow } from 'enzyme';
import * as React from 'react';

import * as renderer from 'react-test-renderer';
import DropdownToggle from '../../../modules/common/components/DropdownToggle';

describe('DropdownToggle component', () => {
  const defaultProps = {
    children: 'child',
    bsRole: 'Role'
  };

  test('renders DropdownToggle successfully', () => {
    shallow(<DropdownToggle {...defaultProps} />);
  });

  test('snapshot matches', () => {
    const rendered = renderer
      .create(<DropdownToggle {...defaultProps} />)
      .toJSON();

    expect(rendered).toMatchSnapshot();
  });

  test('renders successfully with default value', () => {
    const wrapper = mount(<DropdownToggle {...defaultProps} />);
    const props = wrapper.props();

    expect(props).toMatchObject(defaultProps);
  });
});
