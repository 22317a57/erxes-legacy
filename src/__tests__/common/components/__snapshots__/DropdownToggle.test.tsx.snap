// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownToggle component snapshot matches 1`] = `
<div
  onClick={[Function]}
>
  child
</div>
`;
